### DISCUSSION TITLE
Adapt log example to use OldestFirst sorting for -r/--reverse option

### DISCUSSION THREAD

**Discussion by jakewilliami** [GitoxideLabs/gitoxide#1669] [Q&A] [answered]
How can I adapt [the log example](https://github.com/GitoxideLabs/gitoxide/blob/ccd6525c/examples/log.rs) to reverse the order of the `rev_walk` by commit time without collecting the iterator, reversing it, and turning it back into an iterator?

I tried this:
```rust
// ...
use gix::{
    // ...
    traverse::commit::simple::CommitTimeOrder,
};
// ...
    let sorting = if args.breadth_first {
        Sorting::BreadthFirst
    } else {
        Sorting::ByCommitTime(if !args.reverse {
            CommitTimeOrder::NewestFirst
        } else {
            CommitTimeOrder::OldestFirst
        })
    };
// ...
```

Rather than doing it in a vector:
```rust
    if args.reverse {
        let mut results: Vec<_> = log_iter.collect();
        results.reverse();
        log_iter = Box::new(results.into_iter());
    }
```

But this produces incorrect results.

**Comment by Byron** [comment, marked as answer]
There is no way to reverse the order of commits shown other than collecting them first.

`CommitTimeOrder::OldestFirst` has a different effect as it sorts only the commits at each 'junction' of the graph, which can be used to reach the beginning of the graph much faster. But that's already it.

**Reply by jakewilliami** [reply]
Thanks Sebastian, I didn't realise that.
